receiver/otlpreceiver/otlphttp.go

```diff
@@ -28,3 +35,44 @@ type xProtobufMarshaler struct {
 func (*xProtobufMarshaler) ContentType() string {
 	return "application/x-protobuf"
 }
+
+var jsonMarshaller = &jsonpb.Marshaler{}
+
+// OTLPErrorHandler encodes the HTTP error message inside a rpc.Status message as required
+// by the OTLP protocol.
+func OTLPErrorHandler(w http.ResponseWriter, r *http.Request, errMsg string, statusCode int) {
+	var (
+		msg []byte
+		s   *status.Status
+		err error
+	)
+	// Pre-computed status with code=Internal to be used in case of a marshaling error.
+	fallbackMsg := []byte(`{"code": 13, "message": "failed to marshal error message"}`)
+	fallbackContentType := "application/json"
+
+	switch statusCode {
+	case http.StatusBadRequest:
+		s = status.New(codes.InvalidArgument, errMsg)
+	default:
+		s = status.New(codes.Internal, errMsg)
+	}
+
+	contentType := r.Header.Get("Content-Type")
+	switch contentType {
+	case "application/json":
+		buf := new(bytes.Buffer)
+		err = jsonMarshaller.Marshal(buf, s.Proto())
+		msg = buf.Bytes()
+	default:
+		msg, err = proto.Marshal(s.Proto())
+	}
+	if err != nil {
+		msg = fallbackMsg
```

Comment:
Had trouble figuring out how to test this flow. As errMsg is a string, there shouldn't be a condition in which there would be a marshalling error. 
Do I ignore the error to make the codecov happy?

Reply:
That's OK, no need to change anything. It is OK that we know some code paths are not reachable and don't have test coverage.